excel.go

```diff
@@ -0,0 +1,258 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"sort"
+
+	"github.com/xuri/excelize/v2"
+)
+
+type PasswordRow struct {
+	Password string
+	Row      int
+}
+
+func toSheetCoord(coord int) int {
+	return coord + 1
+}
+
+func isValid(f *excelize.File, sheet string, cols []string, rowNum, usernameXCoord, passwordXCoord int) (bool, error) {
+	// check if username or password is empty
+	username, err := getCellValue(f, sheet, usernameXCoord, rowNum)
+	if err != nil {
+		return false, err
+	}
+	password, err := getCellValue(f, sheet, passwordXCoord, rowNum)
+	if err != nil {
+		return false, err
+	}
+
+	return username != "" && password != "", nil
+
+}
+
+func getHeaderToXCoord(headerRow []string) map[string]int {
+	headerToXCoord := make(map[string]int, len(headerRow))
+	for i, cell := range headerRow {
+		headerToXCoord[cell] = i
+	}
+	return headerToXCoord
+}
+
+func getMACFinUsers(f *excelize.File, input *Input) (map[string]string, error) {
+	rows, err := f.GetRows(input.SheetName)
+	if err != nil {
+		return nil, fmt.Errorf("failed getting rows from %s in %s: %s", input.SheetName, input.Filename, err)
+	}
+
+	users := make(map[string]string)
+
+	headerToXCoord := getHeaderToXCoord(rows[0])
+	usernameXCoord := headerToXCoord[input.UsernameHeader]
+	passwordXCoord := headerToXCoord[input.PasswordHeader]
+	rowOffset := input.RowOffset
+
+	for i, row := range rows[rowOffset:] {
+		valid, err := isValid(f, input.SheetName, row, i+rowOffset, usernameXCoord, passwordXCoord)
+		if err != nil {
+			log.Printf("error validating row %d: %s", toSheetCoord(i+rowOffset), err)
+			continue
+		}
+		if !valid {
+			continue
+		}
+
+		users[row[usernameXCoord]] = row[passwordXCoord]
+	}
+
+	return users, nil
+}
+
+func getManagedUsers(f *excelize.File, input *Input) (map[string]PasswordRow, error) {
```

Comment:
To answer a question you asked before: 
> However, I am not using named return values. Is there a benefit to doing that?
In my mind the only benefit is if it is unclear with the names what the return values mean

Reply:
Ok. I left this as is.